vita3k/host/src/pup.cpp

```diff
@@ -0,0 +1,473 @@
+// Vita3K emulator project
+// Copyright (C) 2018 Vita3K team
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include <crypto/aes.h>
+#include <host/pup_types.h>
+#include <miniz.h>
+#include <util/fs.h>
+#include <util/string_utils.h>
+
+#include <fstream>
+
+std::map<int, std::string> pup_types = {
```

Comment:
```suggestion
static std::map<int, std::string> PUP_TYPES = {
```

@VelocityRa Do we need to uppercase this variable? (it's global)